internal/controller/databaseclusterrestore_controller.go

```diff
@@ -451,8 +451,15 @@ func (r *DatabaseClusterRestoreReconciler) restorePG(ctx context.Context, restor
 	if err != nil && !k8serrors.IsNotFound(err) {
 		return err
 	}
+	defer func() {
+		updErr := r.updateStatusPG(ctx, restore, pgCR)
+		if updErr != nil {
```

Comment:
it is better to return error so that in case there is really error K8S will re0send event again and status may be updated successfully.